redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -25,22 +26,8 @@ sealed trait Output[+A] {
 
   private[redis] final def unsafeDecode(respValue: RespValue)(implicit codec: Codec): A =
     respValue match {
-      case RespValue.Error(msg) if msg.startsWith("ERR") =>
-        throw RedisError.ProtocolError(msg.drop(3).trim)
-      case RespValue.Error(msg) if msg.startsWith("WRONGTYPE") =>
-        throw RedisError.WrongType(msg.drop(9).trim)
-      case RespValue.Error(msg) if msg.startsWith("BUSYGROUP") =>
-        throw RedisError.BusyGroup(msg.drop(9).trim)
-      case RespValue.Error(msg) if msg.startsWith("NOGROUP") =>
-        throw RedisError.NoGroup(msg.drop(7).trim)
-      case RespValue.Error(msg) if msg.startsWith("NOSCRIPT") =>
-        throw RedisError.NoScript(msg.drop(8).trim)
-      case RespValue.Error(msg) if msg.startsWith("NOTBUSY") =>
-        throw RedisError.NotBusy(msg.drop(7).trim)
-      case RespValue.Error(msg) =>
-        throw RedisError.ProtocolError(msg.trim)
-      case success =>
-        tryDecode(success)
+      case error: RespValue.Error => throw error.toRedisError
+      case success                => tryDecode(success)
```

Comment:
Was this necessary?

Reply:
I need this logic in RedisClusterExecutorLive to know what error returned